File: solutions/csharp/darts/3/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y)
    {
        // convert coordinates into a distance from the center
     double Distance = Math.Sqrt( x * x + y * y);
        // radius for each circle
        int InnerCircle = 1;
        int MiddleCircle = 5;
        int OuterCircle = 10;

        if (Distance <= InnerCircle)
            return 10;
        
        if (Distance <= MiddleCircle)
            return 5;
        
        if (Distance <= OuterCircle)
            return 1;
        
        else 
            return 0;
    }
}

File: solutions/kotlin/darts/1/src/main/kotlin/Darts.kt
import kotlin.math.sqrt

object Darts {

    // target radius 10 units
    // center radius 1 unit
    // middle ring radius 5 units

    fun score(x: Number, y: Number): Int {
        val distance = calcDistance(x.toDouble(), y.toDouble())
        return when {
            distance <= 1 -> 10
            distance <= 5 -> 5
            distance <= 10 -> 1
            else -> 0
        }
    }
    fun calcDistance(x: Double, y: Double): Double {
        return sqrt(x * x + y * y)
    }
}
